docs/devel/networking.md

```diff
@@ -205,3 +205,136 @@ to the user.
 When the pod networking interface does not feature an IP address, the in-pod
 DHCP server will not be started, leaving the VM with plain L2 connection via
 the in-pod bridge.
+
+### Masquerade binding mechanism
+Similar to the [bridge bind mechanism](#bridge-binding-mechanism), triggering
+the masquerade `BindMechanism` requires a VMI configuration featuring a
+network with the `masquerade` interface type. KubeVirt provides an example of
+a
+[fedora based VMI](https://github.com/kubevirt/kubevirt/blob/master/examples/vmi-masquerade.yaml)
+in the project's examples folder.
+
+The masquerade bind mechanism has plenty in common with bridge binding; both
+have virt-handler create an in-pod bridge, set the pod networking interface as
+its slave, generate a similar looking interface domain xml element - e.g.
+interface type *bridge*, and same *source* and *target* values. Both phases of
+the networking configuration communicate by caching data in the VIF structure.
+
+However, the similarities end there; while the 'networking' infrastructure is
+the same, VM networking works completely different.
+
+In masquerade binding, the goal is to NAT the traffic from the pod interface
+into the VM interface via IPtables / NFtables. Before venturing into details,
+the relevant knobs should be described.
+
+There are 2 knobs that impact the configuration of the masquerade binding:
+  - `ports`: an attribute of the interface, described in the
+    `spec::domain::interfaces` subtree. Here the user indicates the white-list
+     of ports and protocols. It is important to mention that when the list is
+     omitted, all traffic **all** ports will be opened.
+  - `vmNetworkCIDR`: the CIDR from which the in-pod bridge **and** the VM will
+    get their IP address. This attribute is defined in the `spec::networks`
+    subtree. It defaults to `10.0.2.0/24`.
+
+Please refer to the short example below to visualize the aforementioned knobs:
+```yaml
+...
+spec:
+  domain:
+    devices:
+      ...
+      interfaces:
+      - masquerade: {}
+        name: masqueradenet
+        ports:
+        - name: http
+          port: 80
+          protocol: TCP
+        - name: http
+          port: 80
+          protocol: TCP
+  networks:
+  - name: masqueradenet
+    pod:
+      vmNetworkCIDR: 10.11.12.0/24
+```
+
+As with bridge binding, the `discoverPodNetworkInterface` caches the MTU of
+the pod networking interface. It also assigns an IP address from the configured
+CIDR (or the default), and reserves an IP for the gateway - which will be the
+in-pod bridge. Re-using the example above, we would have `10.11.12.1` as
+gateway IP, and `10.11.12.2` as VM IP.
+
+NAT is configured in the masquerade `preparePodNetworkInterfaces` method. In
```

Comment:
How is `masquerade` fits here? I do not understand this: "in the masquerade `preparePodNetworkInterfaces` method".

Reply:
Removed it.